src/util/xeval/builtin_cast.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use tipb::expression::Expr;
+use util::codec::datum::Datum;
+use util::codec::mysql::{Decimal, Time, Duration, DEFAULT_FSP};
+use super::{Evaluator, EvalContext, Result, Error};
+
+fn invalid_type_error(datum: &Datum) -> Result<Datum> {
+    Err(Error::Eval(format!("invalid expr type: {:?}, expect to be int", datum)))
+}
```

Comment:
It would not be the only one. Since the other casting functions are unimplemented, we could leave it this way until later change.